src/snowflake/connector/bind_upload_agent.py

```diff
@@ -39,12 +40,24 @@ def __init__(
             stream_buffer_size: Size of each file, default to 10MB.
         """
         self.cursor = cursor
+        self._stage_name = random_name_for_temp_object(TempObjectType.STAGE)
         self.rows = rows
         self._stream_buffer_size = stream_buffer_size
-        self.stage_path = f"@{self._STAGE_NAME}/{uuid.uuid4().hex}"
+        self.stage_path = f"@{self._stage_name}/{uuid.uuid4().hex}"
+        self._use_scoped_temp_object = (
+            cursor.connection._session_parameters.get(
+                _PYTHON_SNOWPARK_USE_SCOPED_TEMP_OBJECTS_STRING, True
+            )
+            if cursor.connection._session_parameters
+            else True
+        )
```

Comment:
should we default to `False`? snowflake connector users may not be snowpark users
```suggestion
        self._use_scoped_temp_object = (
            cursor.connection._session_parameters.get(
                _PYTHON_SNOWPARK_USE_SCOPED_TEMP_OBJECTS_STRING, False
            )
            if cursor.connection._session_parameters
            else False
        )
```